linode/instancesharedips/framework_resource.go

```diff
@@ -0,0 +1,223 @@
+package instancesharedips
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_instance_shared_ips",
+				IDType: types.Int64Type,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func CreateOrUpdateSharedIPs(
+	ctx context.Context, client *linodego.Client, plan *ResourceModel, diags *diag.Diagnostics,
+) {
+	linodeID := helper.FrameworkSafeInt64ToInt(plan.LinodeID.ValueInt64(), diags)
+	if diags.HasError() {
+		return
+	}
+
+	createOpts := linodego.IPAddressesShareOptions{
+		LinodeID: linodeID,
+	}
+
+	plan.Addresses.ElementsAs(ctx, &createOpts.IPs, false)
+
+	tflog.Debug(ctx, "client.ShareIPAddresses(...)", map[string]interface{}{
+		"options": createOpts,
+	})
+
+	err := client.ShareIPAddresses(ctx, createOpts)
+	if err != nil {
+		diags.AddError(
+			fmt.Sprintf("failed to update ips for linode %d", linodeID),
+			err.Error(),
+		)
+		return
+	}
+	plan.FlattenSharedIPs(linodeID, createOpts.IPs, true, diags)
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	tflog.Debug(ctx, "Create "+r.Config.Name)
+
+	var plan ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx = populateLogAttributes(ctx, plan)
```

Comment:
We probably can use `tflog.SetField(...)` instead of building the function `populateLogAttributes`.

Reply:
Got it. Looks good!

Reply:
Actually, @yec-akamai I think you are right. Sorry about misunderstood the comment..
I will fix it in a future PR.